sklearn/isotonic.py

```diff
@@ -301,13 +319,48 @@ def _build_y(self, X, y, sample_weight, trim_duplicates=True):
                 np.not_equal(y[1:-1], y[:-2]), np.not_equal(y[1:-1], y[2:])
             )
             return X[keep_data], y[keep_data]
+        elif trim_duplicates and self.centered:
+            # Removed unnecessary points according to the Centered Isotoinc
+            # Regression (CIR) method.
+            X, y = self._build_cir_points(X, y, sample_weight)
+            return X, y
         else:
             # The ability to turn off trim_duplicates is only used to it make
             # easier to unit test that removing duplicates in y does not have
             # any impact the resulting interpolation function (besides
             # prediction speed).
             return X, y
 
+    def _build_cir_points(self, X, y, sample_weight):
+        """
+        Generate the points X_, y_ for the output function when Centered
+        Isotonic Regression (CIR) is chosen, based on non-trimmed points X, y.
+        """
+        X_ = y_ = np.array([])
```

Comment:
This is wrong as `X_` and `y_` are now referring to the exact same array.
Anyhow, it is better to either initialize arrays of the correct size or, if that's not possible, work with lists as appending is cheaper and convert to an array at the end.

Reply:
Thanks for the suggestion. I updated this now, do you think the new approach is good?

Reply:
I think so, yes.